packages/@aws-cdk/aws-cloudfront/lib/public-key.ts

```diff
@@ -54,14 +55,15 @@ export class PublicKey extends Resource implements IPublicKey {
   public readonly publicKeyId: string;
 
   constructor(scope: Construct, id: string, props: PublicKeyProps) {
-    super(scope, id, {
-      physicalName: props.publicKeyName ??
-        Lazy.string({ produce: () => this.generateName() }),
-    });
+    super(scope, id);
+
+    if (!Token.isUnresolved(props.encodedKey) && !/^-----BEGIN PUBLIC KEY-----/.test(props.encodedKey)) {
+      throw new Error('Invalid public key.');
```

Comment:
Always good to give a bit more information than you think is necessary; especially once constructs are nested, it can become unclear what the issue is.

```suggestion
      throw new Error(`public key must be in PEM format (with the BEGIN/END PUBLIC KEY lines); got ${props.encodedKey}`);
```

Reply:
I'm on it.